server/sonar-server/src/main/java/org/sonar/server/issue/ws/IssueJsonWriter.java

```diff
@@ -80,32 +124,37 @@ public void write(JsonWriter json, Issue issue, Map<String, User> usersByLogin,
     Duration debt = issue.debt();
     Date updateDate = issue.updateDate();
 
-    json
-      .prop("key", issue.key())
-      .prop("component", file != null ? file.getKey() : null)
-      // Only used for the compatibility with the Issues Java WS Client <= 4.4 used by Eclipse
-      .prop("componentId", file != null ? file.getId() : null)
-      .prop("project", project != null ? project.getKey() : null)
-      .prop("subProject", subProject != null ? subProject.getKey() : null)
-      .prop("rule", issue.ruleKey().toString())
-      .prop("status", issue.status())
-      .prop("resolution", issue.resolution())
-      .prop("severity", issue.severity())
-      .prop("message", issue.message())
-      .prop("line", issue.line())
-      .prop("debt", debt != null ? durations.encode(debt) : null)
-      .prop("assignee", issue.assignee())
-      .prop("reporter", issue.reporter())
-      .prop("author", issue.authorLogin())
-      .prop("actionPlan", actionPlanKey)
-      .prop("creationDate", isoDate(issue.creationDate()))
-      .prop("updateDate", isoDate(updateDate))
-      .prop("closeDate", isoDate(issue.closeDate()));
-
-    writeTags(issue, json);
-    writeIssueComments(commentsByIssues.get(issue.key()), usersByLogin, json);
-    writeIssueAttributes(issue, json);
-    writeIssueExtraFields(issue, usersByLogin, actionPlanByKeys, extraFields, json);
+    json.prop(FIELD_KEY, issue.key());
+    JsonWriterUtils.writeIfNeeded(json, file != null ? file.getKey() : null, FIELD_COMPONENT, fields);
```

Comment:
I don't think so, ``JsonWriter`` is only about - well, writing to JSON. Knowing if a field should be written or not seems like another concern to me.